### ISSUE TITLE
🌱 Investigate using KubeadmControlPlane (KCP) for NestedCluster

### ISSUE THREAD

**Issue by christopherhein** [kubernetes-sigs/cluster-api-provider-nested#44]
## TL;DR
This is a spike to investigate what it would be like to use KCP for implementing CAPN. https://github.com/kubernetes-sigs/cluster-api/tree/master/controlplane/kubeadm KCP uses Kubeadm's `ClusterConfiguration` object and is what nearly all Cluster API providers use, this was originally not gone with because it only supports cloudinit based deployments. 

## Background:
KCP needs to map to `Machines`, `Machines` have to map to actual Kubernetes `Nodes` and the control plane pods need to show up on those specific. KCP also needs to be able to `exec` from the management cluster into the workload cluster to get access to `etcd` for health checks. KCP also doesn't manage client certificates, these are handled on each node by `kubeadm` that is called through `cloudinit`. We should look at the in-tree `docker` provider for some inspiration on how we could use the `kcp` outputs to create Pod based control planes within custom `Machine` controllers. https://github.com/kubernetes-sigs/cluster-api/tree/master/test/infrastructure/docker

**Comment by MadhavJivrajani** [comment]
/remove-kind design
/kind feature
`kind/design` is migrated to `kind/feature`, see https://github.com/kubernetes/community/issues/6144 for more details
